### ISSUE TITLE
Container Seems to only be using one processor (thread)

### ISSUE THREAD

**Issue by TopCheddar27** [haugene/docker-transmission-openvpn#1461]
It seems to me that after the 3.0 update to alpine, something has gone amiss with the transmission deamon. It is now constantly eating a full core during only a 50 megabit download. And since it seems the actual transmission process is single threaded, it causes extreme instability with both the HTML serving and with downloads/uploads. 

I launch, update, and troubleshoot my images through docker-compose. My compose file (with redactions) is below:

"version: "3.3"

services:
  openvpn-trans:
    container_name: openvpn-trans
    volumes:
      - '/*********:/data'
      - '/etc/localtime:/etc/localtime:ro'
    environment:
      - PUID=1000
      - PGID=1000
      - OPENVPN_PROVIDER=PIA
      - OPENVPN_CONFIG=Spain
      - OPENVPN_USERNAME=**********
      - OPENVPN_PASSWORD=**********
      - WEBPROXY_ENABLED=false
      - LOCAL_NETWORK=192.168.0.0/16,10.8.0.0/16
      - TRANSMISSION_DOWNLOAD_DIR="/data/Downloads"
      - TRANSMISSION_SPEED_LIMIT_UP=800
      - TRANSMISSION_SPEED_LIMIT_UP_ENABLED=true
      - TRANSMISSION_WATCH_DIR_ENABLED=false
      - TRANSMISSION_CACHE_SIZE_MB=512
    cap_add:
      - NET_ADMIN
    logging:
      driver: json-file
      options:
          max-size: 10m
    ports:
      - '9091:9091'
    image: haugene/transmission-openvpn
    restart: on-failure"

LOGS -- It starts normally and connects with a forwarded port just as before. It accomplishes this task. It is just that for some reason its blasting its main transmission process to use a full core, and slows all container processes down.

Ubuntu Server 20.04 Headless

Docker 19.03.13 stable

What HTOP looks like with a 20 Mbit download:
![image](https://user-images.githubusercontent.com/73973868/98188112-6d49e900-1ee0-11eb-9b62-5348e24fe662.png)

**Comment by phasma343** [comment]
Seems to be  multi-threaded on Synology, and on my Fedora 33 Server VM as well.

Does it do the same under the dev tag?

**Comment by TopCheddar27** [comment]
Sorry for the late response. Have gone through a migration and wanted to confirm that this this is still happening on new hardware. And it is.

Same problem as before. It's also worth noting that I have a 500 megabit down connection with no QOS or funky throughput bottlenecks routing side. I can load a qbittorrent container and it will fully utilize all threads available to the system. I can easily download 4 large files with a total combined throughput of  around 300 megabit, and even then saturation of the processor is <50% on all threads. This is how this container used to work before the change in both the base image and the backen network stack.

So that is what I am leaning towards. I think somehow the change over to the wireguard stack is making the decryption single threaded. I could be completely off base though. Im not totally sure if the default PIA network stack has been fully changed over to the wireguard for the next gen port forwarding servers. 

I also may be missing a switch somewhere docker side that makes this container run only on one core. However that would be defined in the compose file (as I do that with multiple other containers such as proxies). 

Any help on this would be appreciated because it seems that either A) I am the only one experiencing this B) Nobody has actually noticed the performance degradation. All config files above are still applicable and HTOP process saturation looks the exact same now as it did then. 

This is running on local hardware btw. On ZFS pools with appropriate caching mechanisms. So I KNOW that the bottleneck is not storage interface side.

**Comment by TopCheddar27** [comment]
Have read through some Transmission GitHub open issues and it seems some others are having trouble with the IO pipeline recently as well. I can confirm this by noticing normal speeds at the start when cache is filling. Then once filled speeds nosedive and WebUI responsivness becomes less than optimal. Im saying nosedive because we are looking at 30 MBs down to maybe 400 KBs down. IOTOP does not show excessive IO throughput from the process either. 

I also made sure that network translation was not the problem. Running on the host interface with no port or ip tagging yields no better results. I also figured out that OpenVPN is used and is not the the problem. Encryption overhead from the container is not even close to a problem on my processor. 

All I know is around October / Novemeber, when Haugene incorporated the newest alpine based build from Transmission, the IO and CPU bottlenecking started. I may be better off bringing this up over there, however it seems like developer responsiveness is... lackluster. 

I may just go down the grueling route of setting up my own image that incorporates qbittorrent with a Wireguard connection. The problem is that it needs to also run VPN port request every 20 minutes or so otherwise port forwarding will bomb out.
